core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -378,11 +378,8 @@ struct HIPParallelLaunchKernelInvoker<DriverType, LaunchBounds,
                             dim3 const &block, int shmem,
                             HIPInternal const *hip_instance) {
     DriverType *driver_ptr = reinterpret_cast<DriverType *>(
-        hip_instance->scratch_functor(sizeof(DriverType)));
-
-    hipMemcpyAsync(driver_ptr, &driver, sizeof(DriverType), hipMemcpyDefault,
-                   hip_instance->m_stream);
-
+        hip_instance->stage_functor_for_execution(
```

Comment:
Does all of this need a lock, as in the constant memory launcher to avoid multiple threads stomping on the same resources?

Reply:
I had the same thought when we discussed it to Bruno but (wrongly) assumed it was locked from the call site.
Good catch